firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/internal/InstallActivity.java

```diff
@@ -53,23 +60,33 @@ protected void onResume() {
 
     if (!isUnknownSourcesEnabled()) {
       // See comment about install progress above. Same applies to unknown sources UI.
-      if (unknownSourceEnablementInProgress) {
+      if (alertDialog.isShowing()) {
         LogWrapper.getInstance()
             .e(
                 TAG
                     + "Unknown sources enablement was already in progress. It was either canceled or failed");
+        alertDialog.dismiss();
         finish();
         return;
       }
-
-      unknownSourceEnablementInProgress = true;
+      LogWrapper.getInstance().d("SHOWING UNKNOWN SOURCES ");
       showUnknownSourcesUi();
       return;
     }
 
     startAndroidPackageInstallerIntent();
   }
 
+  protected void onDestroy() {
+    super.onDestroy();
+    if (alertDialog.isShowing()) {
+      alertDialog.dismiss();
```

Comment:
This prevents the window leak error. My guess is once we deal with the activity and rotations this may be unnecessary, but this will get rid of the error in the meantime.

Reply:
Would you be able to give me a TL;DR on the window leak error?